### PR TITLE
Release tool improvments

### PR DESCRIPTION
**Pull request by davidvossel**
- Adds `#` character to PRs in release notes so they appear as links. 

- Fixes issue with branch cache. We need to clear
  branch cache and re-generate in the event a new branch
  is created

- Fixes issue with branching prow configs. Only attempt
  to pull the latest branch code for the prow config if
  the branch already exists.

### CODE REVIEW

File: releng/release-tool/release-tool.go [hunk 1]
```diff
@@ -386,17 +395,20 @@ func (r *releaseData) forkProwJobs() error {
 	gitbranch := fmt.Sprintf("%s_%s_%s_configs", r.org, r.repo, r.newBranch)
 
 	_, err := gitCommand("-C", r.infraDir, "checkout", "-b", gitbranch)
-	// if err, then branch probably already exists
+
+	// if err, then branch likely already exists
```
Comment (rmohr) [inline releng/release-tool/release-tool.go:399]: Could this potentially lead to a situation where we take the branch from a not-anymore correct commit? Would it make sense to use `-B` instead. Basically create-if-necessary and set to the current commit?

Comment (davidvossel) [reply releng/release-tool/release-tool.go]: yep, fixed

**Comment by davidvossel** [comment]
@rmohr this should be good for another look

**Pull request merged**